proposed/2023/NuGetAudit-without-nuget.org-as-packageSource.md

```diff
@@ -0,0 +1,190 @@
+Additionally, in the case where NuGet doesn't otherwise need to communicate with any package source, checking the package sources for a VDB might trigger a credential provider.
+
+## Drawbacks
+
+<!-- Why should we not do this? -->
+Customers who appear functionally network isolated from nuget.org (for example, by blocking nuget.org at the firewall) will still not be able to use this feature.
+
+## Rationale and alternatives
+
+<!-- Why is this the best design compared to other designs? -->
+<!-- What other designs have been considered and why weren't they chosen? -->
+<!-- What is the impact of not doing this? -->
+
+### Wait for other servers to implement the VulnerabilityInfo resource
+
+While the NuGet team may be able to influence other teams and encourage them to work on implementing support for the `VulnerabilityInfo` resource, ultimately the teams that implement those NuGet servers have their own backlogs and priorities.
+Therefore, the NuGet team doesn't have agency to ensure that it gets implemented, and customers who want to use *NuGetAudit* may not have enough influence to get the implementation prioritized in a timely manor.
+
+Additionally, not all customers use a server that supports upstreaming, or have configured their server to use nuget.org as an upstream source.
+An example is developers working in highly regulated industries where packages need to be individually vetted and approved before it can be used.
+The reasons customers may do this do not preclude them from using nuget.org as an audit source, so allowing these customers an easy way to use *NuGetAudit* benefits them.
+
+Therefore, in order to provide customers with the ability to use NuGetAudit most easily, it's more effective to implement a new feature that does not depend on other teams to implement features.
+
+### NuGet Team supplied upsourcing server
+
+The NuGet team could provide a web application that looks like a NuGet package source that never contains any packages (and doesn't support push), and the `VulnerabilityInfo` resource effectively proxies nuget.org's VDB.
+However, getting permission to install a new web app is likely difficult in most corporate environments, so wouldn't be practical.
+
+Besides the high up-front cost of implementation, it would also have ongoing maintenance costs to the NuGet team.
+For example ensuring it's patched following every Patch Tuesday, and probably maintaining docker images to make it easy to use.
+
+Unlike this spec's proposal, this vulnerability-only upsourcing server idea could in theory enable customers in companies that block nuget.org at the firewall to get always up-to-date VDB.
+But in such a restrictive environment, I expect it's even less likely that their security would allow this web app to be installed than other companies that simply have a policy of "don't use nuget.org as a package source".
```

Comment:
Another idea is to a static source that just points to nuget.org. 

That doesn't help the people blocking with firewalls, but it helps the people that just don't want to use nuget.org